### PR TITLE
Bring next release candidate up to date w/ master

### PR DESCRIPTION
**Pull request by RKuttruff**
- Ensured links in the `/job` endpoint are https
- SDAP-488: Workaround to build issue on Apple Silicon (M1/M2). Image build installs nexusproto through PyPI instead of building from source. A build arg `BUILD_NEXUSPROTO` was defined to allow building from source if desired

### CODE REVIEW

File: analysis/webservice/algorithms/doms/ExecutionStatus.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def calc(self, request, **args):
                 )
 
         job_status = NexusExecutionResults.ExecutionStatus(execution_details['status'])
-        host = f'{request.requestHandler.request.protocol}://{request.requestHandler.request.host}'
+        host = f'https://{request.requestHandler.request.host}'
```
Comment (skorper) [inline analysis/webservice/algorithms/doms/ExecutionStatus.py:57]: What's the reason for this change?

Comment (RKuttruff) [reply analysis/webservice/algorithms/doms/ExecutionStatus.py:57]: @skorper This was an issue with CDMS. The links for NetCDF and CSV result downloads were being blocked by Chrome or something, and when they were downloaded with right click and save as they were quarantined or something requiring additional confirmation to be able to be used.
We noticed all the links were `http://` links and figured it was a security feature to distrust downloads from unsecured URLs so I forced the links to be `https://`.

**Review by skorper — APPROVED** [review]
